lib/find-emails.js

```diff
@@ -33,28 +33,33 @@ module.exports = function findEmails (args, callback) {
   var timeout = setTimeout(function () {
     if (state.connection === 'virgin') {
       state.errors.push('imap connection error')
-      maybeErrorCallback()
     } else if (state.connection === 'open') {
       state.errors.push('connection closing in timeout error')
       args.log.error('timeout did not close imap connection in time')
-      maybeErrorCallback()
-      maybeErrorCallback = function () {
-        args.log.error('callback called after timeout')
-      }
+      imapDisconnect()
     } else if (state.connection === 'closed' && !state.queue.idle()) {
       state.errors.push('attachment handling in timeout error')
       args.log.error('attachments could not be handled in time')
-      maybeErrorCallback()
-      maybeErrorCallback = function () {
-        args.log.error('callback called after timeout')
-      }
+    } else {
+      args.log.error('timeout hit unmet case')
+      imapDisconnect()
+    }
+
+    maybeErrorCallback({abort: true})
+
+    maybeErrorCallback = function () {
```

Comment:
the abort: true or the replacing of the callback? :-D

it s pretty ugly, i agree